src/osdc/Objecter.h

```diff
@@ -688,21 +696,18 @@ struct ObjectOperation {
       if (r >= 0) {
 	auto p = bl.cbegin();
 	try {
-	  if (pattrs)
```

Comment:
This checks whether the pointer is null and doesn't decode to it if so. You found a bug, but the right fix is to get rid of the uses of `&ignore` and pass in a null pointer instead.

Reply:
To be fair, the compiler found the bug :)
I simplified the use of the temp by keeping it in the object. There's probably a more elegant way to do this but it would involve a much greater change